src/evaluation/move_vote_map.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "movegen/move_types.h"
+
+namespace evaluation {
+
+template<size_t maxSize>
+class MoveVoteMap {
+
+    using MoveVotePair = std::pair<movegen::Move, uint8_t>;
+
+public:
+    MoveVoteMap() = default;
+
+    MoveVoteMap(std::initializer_list<MoveVotePair> init)
+    {
+        assert(init.size() < maxSize);
+
+        for (auto& entry : init) {
+            m_entries.at(m_size++) = std::move(entry);
+        }
+    }
+
+    constexpr void insertOrIncrement(movegen::Move ext_move, uint8_t ext_vote)
+    {
+        bool matchFound = false;
+        for (auto& [move, vote] : *this) {
+            if (move == ext_move) {
+                matchFound = true;
+                vote += ext_vote;
+            }
+        }
+
+        if (!matchFound) {
+            assert(m_size < maxSize);
+            m_entries.at(m_size++) = MoveVotePair(ext_move, ext_vote);
+        }
+    }
+
+    constexpr void clear()
+    {
+        for (auto& entry : *this) {
+            entry = MoveVotePair {};
+        }
```

Comment:
this is, in theory, not necessary. 
I don't have strong opinions about it, but it *should* be enough to just reset the size :)

Reply:
I added this for cleanness, because it makes debugging etc easier. But since the voting map isn't complicated right now, I agree that we can remove it.